librz/type/base.c

```diff
@@ -174,65 +174,16 @@ RZ_API void rz_type_db_save_base_type(const RzTypeDB *typedb, const RzBaseType *
 /**
  * \brief Returns C representation as string of RzBaseType
  *
- * \param typedb Type Database instance
- * \param type RzBaseType to convert
+ * \param typedb type database instance
+ * \param btype RzBaseType to convert
+ * \param opts options for pretty printing (see RzTypePrintOpts)
+ * \param unfold_level level of unfolding to do in case of nested structures/unions (any negative number means maximum unfolding, i.e. INT32_MAX. 0 means no unfolding, just the typename and identifier, if any)
  */
-RZ_API RZ_OWN char *rz_type_db_base_type_as_string(const RzTypeDB *typedb, RZ_NONNULL const RzBaseType *type) {
-	rz_return_val_if_fail(typedb && type && type->name, NULL);
+RZ_API RZ_OWN char *rz_type_db_base_type_as_pretty_string(RZ_NONNULL const RzTypeDB *typedb, RZ_NONNULL const RzBaseType *btype, unsigned int opts, int unfold_level) {
```

Comment:
Instead of changing the function I suggest to make it a new function, and keep `rz_type_db_base_type_as_string()` API, it's useful for simple cases, just internally it will call the `rz_type_db_base_type_as_pretty_string()` with some predefined parameters.